managed/models/files_model.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/pkg/errors"
+)
+
+//go:generate ../../bin/reform
+
+// File represents File as stored in database.
+//
+//reform:files
+type File struct {
+	Name      string    `reform:"name,pk"`
+	Content   []byte    `reform:"content"`
+	UpdatedAt time.Time `reform:"updated_at"`
+}
+
+// InsertFileParams represent insert file params. Call validate before use.
+type InsertFileParams struct {
+	Name    string
+	Content []byte
+}
+
+func checkFileName(name string) error {
+	switch want := govalidator.SafeFileName(name); {
+	case name == ".", name == "..":
+		fallthrough
+	case len(want) == 0:
+		return fmt.Errorf(`invalid file name: %s`, name)
+	case name != want:
+		return fmt.Errorf(`invalid file name: %s; want: %s`, name, want)
+	}
+	return nil
+}
+
+// Validate validates params.
+func (p *InsertFileParams) Validate() error {
+	if err := checkFileName(p.Name); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateFileParams represent update file params.
+type UpdateFileParams struct {
```

Comment:
But what benefits does it provide?
As far as I understand, only saving lines of code?

It's just that this approach has proven itself to have a number of advantages.
By separating in this way, we can:
1. Encapsulate the validation logic required for the update and insert separately.
2. Combining this under a common interface, we can use this structure in other validating libraries.
3. You do not need to write a validation for the combined structure in which there will be many branches.
4. This will make the code readable and extensible in the future.